### DISCUSSION TITLE
Getting "Kotlin: Unresolved reference: scene2d" error

### DISCUSSION THREAD

**Discussion by jgandert** [libktx/ktx#310] [Q&A] [answered]
Any ideas what could be causing this?

**Comment by czyzby** [comment]
9/10 problems like this are caused by Gradle setup - a `clean` and `build` should help.

**Comment by jgandert** [comment]
It was this issue again: https://youtrack.jetbrains.com/issue/KT-32047

IDEA didn't show any problems, but when trying to run the project, there were `Kotlin: Unresolved reference: ` everywhere. 

I've deleted all the `build` and `out` directories and went back to the previous KTX version. Haven't yet tried to update again. 

Managing dependencies and upgrading when there are breaking changes is definitely my absolute least favorite thing in software development.

**Comment by czyzby** [comment]
Sorry to hear that. However, it seems that it's not as much of a breaking changes problem as a problem with your local setup. Deleting Gradle and IDE related folders is what I would have done too, or just building from a clean copy of the repository.

**Comment by jgandert** [comment]
@czyzby Now I have reset everything to the previous state, deleted `~./gradle`, let IDEA set up everything again and suddenly I'm getting `Ensure that you have a dependency on the Kotlin standard library` errors  everywhere, despite being on `1.4.10` (which supposedly doesn't need that dependency) and having `implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk8:$kotlinVersion"` - Any ideas? I'm starting to believe that this project is haunted.

**Comment by czyzby** [comment]
Which Gradle version are you on? If you have set up your project a while back with the official tool, you might be on a very old version that isn't fully supported by IDEA.

To be honest, that dependency should be enough. Are you sure it's in the core project?

I can give it a try locally if you can post a reproducible example, but usually a rebuild is all I need when I encounter issues like that. Have you tried [cleaning IDEA cache](https://stackoverflow.com/questions/17561826/how-to-clean-project-cache-in-intellij-idea-like-eclipses-clean)?

**Comment by jgandert** [comment]
I honestly think this is an IntelliJ / Kotlin problem, so definitely not your fault. Look at this beauty:

![Screenshot_20201023_222629](https://user-images.githubusercontent.com/3063858/97051100-e2332f80-157e-11eb-8dc4-0bd5780e8580.png)

from `File > Project Settings`

I don't even know where that jdk7 comes from. I've always used JDK 8. Coroutines? I'm not using them. There's zero mention of coroutines in the whole codebase.

Oh.. and now IntelliJ IDEA froze. (IDEA = latest version)

> Have you tried cleaning IDEA cache?

Thanks, yes, but I'll try again.

> Which Gradle version are you on?

I'm on gradle 5.4.1

**Comment by czyzby** [comment]
Depending on the module, KTX could define coroutines as a provided dependency (that you have to define manually), but they're marked with `-async` - if you don't use any of these, I don't see how KTX would depend on them at all. I'll be honest, I'm not sure what's going on here. Have you tried setting it up on a different computer?

**Comment by czyzby** [comment]
Do a clean install in a virtual machine. Make sure it's not a problem with your config. :)

**Comment by jgandert** [comment, marked as answer]
Thanks for the suggestions! 

I've deleted the `.idea` folder and re-imported in IDEA from the `.gradle` file and now it's running again. Yay!

I also tried the new plugin syntax [here](https://kotlinlang.org/docs/reference/using-gradle.html#targeting-the-jvm) for Kotlin, but it seems everything broke in the LibGDX project when I did that. (Could be unrelated though.)

I'll try to update KTX again tomorrow!

**Comment by czyzby** [comment]
Glad you worked it out!

There might have been some breaking changes, depending on the module and how old KTX you're currently at, but you should find everything you need in the changelog. The modules with any major changes (like `scene2d`) have their specific migration guides in readme files. Let me know if you encounter any issues.

**Comment by jgandert** [comment]
Thanks for the support! The update went through without a hitch.

No idea  what happened there yesterday. The upside is that all the IntelliJ IDEA stuff is now in the `.idea` folder. (Previously I've had multiple `iml` files inside the core and desktop folders, which I've removed after checking that they haven't been used, and now they're stored in `.idea/modules`.)
